specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/ContainerAppsBuilds.json

```diff
@@ -0,0 +1,403 @@
+            "$ref": "./examples/ContainerAppsBuilds_Delete.json"
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "ContainerAppsBuildConfiguration": {
+      "type": "object",
+      "properties": {
+        "baseOs": {
+          "type": "string",
+          "description": "Base OS used to build and run the app."
+        },
+        "platform": {
+          "type": "string",
+          "description": "Platform to be used to build and run the app."
+        },
+        "platformVersion": {
+          "type": "string",
+          "description": "Platform version to be used to build and run the app."
+        },
+        "environmentVariables": {
+          "type": "array",
+          "items": {
+            "$ref": "./CommonDefinitions.json#/definitions/EnvironmentVariable"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "description": "List of environment variables to be passed to the build, secrets should not be used in environment variable."
+        },
+        "preBuildSteps": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PreBuildStep"
+          },
+          "x-ms-identifiers": [],
+          "description": "List of steps to perform before the build."
+        }
+      },
+      "description": "Configuration of the build."
+    },
+    "ContainerAppsBuildProperties": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/BuildProvisioningState",
+          "description": "Build provisioning state.",
+          "readOnly": true
+        },
+        "buildStatus": {
+          "$ref": "#/definitions/BuildStatus",
+          "description": "Status of the build once it has been provisioned.",
+          "readOnly": true
+        },
+        "destinationContainerRegistry": {
+          "$ref": "#/definitions/ContainerRegistryWithCustomImage",
+          "description": "Container registry that the final image will be uploaded to."
+        },
+        "configuration": {
+          "$ref": "#/definitions/ContainerAppsBuildConfiguration",
+          "description": "Configuration of the build."
+        },
+        "logStreamEndpoint": {
+          "type": "string",
+          "description": "Endpoint from which the build logs can be streamed.",
+          "readOnly": true
+        }
+      },
+      "description": "The ContainerAppBuild properties."
+    },
+    "ContainerAppsBuildResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ContainerAppsBuildProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
```

Comment:
do we need "create" if there is no way to do PUT on this resource?

Reply:
Strictly speaking, I'd think not...